include/rmm/mr/device/detail/arena.hpp

```diff
@@ -39,9 +39,7 @@
 #include <optional>
 #include <set>
 
-namespace RMM_NAMESPACE {
-namespace mr::detail::arena {
-
+namespace rmm::mr::detail::arena {
```

Comment:
This changes the `__attribute__((visibility))` of this namespace from `"default"` to `"hidden"`. Is this deliberate, or an inadvertent editor-induced change?

Reply:
It's deliberate.

Reply:
Ah, OK, can we just hide that one symbol? Add `RMM_HIDDEN` for that symbol.

See discussion in https://github.com/rapidsai/rmm/issues/1652, particularly, we need to be careful about visibility of objects that might make it in to the public namespace.

Reply:
Thank you for sharing. I will check whether the exception can be thrown from there. Considering that the namespace is `detail` and is only used by the MR, which is exposed in the public, it should be fine. That said, I will double check.
Who doesn't love c++ linking, so much fun.

Reply:
@wence- Did a quick test with XGBoost using rmm by inserting a `RMM_EXPECT(false, "test")` in allocate, the exception can go through the shared objects.
```
    _check_call(ret)
  File "/home/jiamingy/workspace/xgboost_dev/xgboost/python-package/xgboost/core.py", line 299, in _check_call
    raise XGBoostError(py_str(_LIB.XGBGetLastError()))
xgboost.core.XGBoostError: [08:30:53] /home/jiamingy/workspace/xgboost_dev/xgboost/src/common/device_vector.cu:23: Memory allocation error on worker 0: RMM failure at: /home/jiamingy/.anaconda/envs/xgboost_dev_125/include/rmm/mr/device/detail/arena.hpp:588: test
```

Reply:
I am confused. Have we just exposed the whole file, or can you just unhide parts of it?

Reply:
We have exposed the memory resource but not the arena. As long as the exception goes through the memory resource them it should go through the shared object.

Reply:
The visibility is controlled by namespace instead of translation unit.

Reply:
oh I had it backwards. I thought this change was exposing more, but it's actually hiding more.

Reply:
May I suggest that we make the hiding explicit?

```
namespace RMM_HIDDEN rmm::mr::detail::arena {
```
?

Reply:
I'm totally happy to make the change. But do you want to do it for the entire repository or do you want me to make the change to this specific file? Or maybe the CMake hidden visibility option is sufficient?

Reply:
Switched to hiding only the spdlog function as suggested offline.